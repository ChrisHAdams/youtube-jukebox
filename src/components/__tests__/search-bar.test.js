import React from "react"
import renderer from "react-test-renderer"
import {render, fireEvent, cleanup} from 'react-testing-library'
import SearchBar from "../search-bar";
import {toHaveStyle} from 'jest-dom';

expect.extend({toHaveStyle})

describe("SearchBar", () => {

  it("renders correctly", () => {

    const tree = renderer
      .create(<SearchBar
                searchValue={""}
                loggedIn={false}
                searchOnClick={"callback"}
                resetOnClick={"callback"}
                themeChange={"callback"}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})


