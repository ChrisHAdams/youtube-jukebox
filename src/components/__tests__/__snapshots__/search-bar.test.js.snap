// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="sc-ifAKCX fFOdlp"
  data-testid="searchBar"
>
  <div>
    <input
      className="sc-bdVaJa fIgOID"
      data-testid="searchField"
      disabled={true}
      name="searchField"
      placeholder="Search for videos"
      type="text"
      value=""
    />
  </div>
  <button
    className="sc-bwzfXH jcuoYi"
    data-testid="searchButton"
    onClick="callback"
  >
    Search
  </button>
  <button
    className="sc-htpNat fBDJQi"
    onClick="callback"
  >
    Reset
  </button>
  <button
    className="sc-bxivhb dsnugm"
    data-testid="themeChangeButton"
    onClick="callback"
  >
    Change Theme
  </button>
</div>
`;
